File: normal/diffLang/pathSum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool hasPathSum(TreeNode* root, int targetSum) {
        if (root == nullptr) return false; // Base case: empty tree.

        // Check if it's a leaf node and the value matches the remaining targetSum.
        if (root->left == nullptr && root->right == nullptr && root->val == targetSum) {
            return true;
        }

        // Recurse for left and right subtrees with the updated targetSum.
        targetSum -= root->val;
        return hasPathSum(root->left, targetSum) || hasPathSum(root->right, targetSum);
    }
};

File: normal/diffLang/pathSum.py
class Solution:
    def hasPathSum(self, root, targetSum):
        if root is None:
            return False  # Base case: empty tree.

        # Check if it's a leaf node and the value matches the remaining targetSum.
        if root.left is None and root.right is None and root.val == targetSum:
            return True

        # Recurse for left and right subtrees with the updated targetSum.
        targetSum -= root.val
        return self.hasPathSum(root.left, targetSum) or self.hasPathSum(root.right, targetSum)
